neqo-transport/src/qlog.rs

```diff
@@ -296,23 +296,18 @@ pub fn packets_lost(qlog: &NeqoQlog, pkts: &[SentPacket], now: Instant) {
 
 pub fn packet_received(
     qlog: &NeqoQlog,
-    public_packet: &PublicPacket,
-    payload: &DecryptedPacket,
+    pt: PacketType,
+    pn: PacketNumber,
+    plen: usize,
+    body: &[u8],
```

Comment:
Would it make sense to build a struct for this chunk of values?  That might help with your clippy warning (which I think should be fixed).